fix(#20): Add `params` option to the mustermann grape pattern for taking into account the parameter type

jcagarcia opened the pull request:
Hey 👋 

I've decided to create this PR for including the `params` options to the mustermann grape pattern. Fixing #20.

I've added several tests and seems that if we provide the proper params information, we are able to generate more precise regular expressions.

```ruby
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"Integer"}}).params('/1') # => { id: '1'}
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"Integer"}}).params('/foo') # => nil
Mustermann.new('/:id', type: :grape, params: {"id"=>{:type=>"String"}}).params('/foo') # => { id: 'foo'}
```

What do you think?

jcagarcia committed:
fix(#20): Add `params` option to the mustermann grape pattern for taking into account the parameter type

dblock reviewed (commented):
I like it! Thanks @jcagarcia.

Let's increment the version to 1.1.0 here since this is a new feature.

@namusyaka, you around?

dblock commented:
As you fix CI, see if it's easy to add a Ruby 3.x, feel free to make separate PRs.

jcagarcia committed:
fix(#20): Upgrade version to 1.1.0

jcagarcia committed:
fix(#20): Run tests with Ruby 3.x

jcagarcia commented on CHANGELOG.md:
```diff
@@ -0,0 +1,10 @@
+### 1.0.3 (Next)
```
Totally welcome!! 👍

jcagarcia commented:
Hi @dblock ,

Thanks a lot for the review. I've updated the version to `1.1.0`. Also, I've added Ruby 3.x to the gitHub action that runs the tests.

Let me know if something else is needed! :)

dblock commented:
Will let this marinate a little and see if we hear from @namusyaka, otherwise I'll merge soon.

jcagarcia commented:
Hi again @dblock 👋 

Do you think we can merge this change and release `1.1.0` version of `mustermann-grape`?

dblock commented:
I released 1.1.0.

dblock commented:
@jcagarcia Want to help us with maintaining mustermann-grape? LMK your rubygems username if you're down.

jcagarcia commented:
> @jcagarcia Want to help us with maintaining mustermann-grape? LMK your rubygems username if you're down.

Sure @dblock ! Really happy to help :) This is my rubygems profile https://rubygems.org/profiles/jcagarcia

dblock commented:
Accept the maintainer invite here: https://github.com/ruby-grape/mustermann-grape/invitations. Added you to Rubygems! Thanks.
